### ISSUE TITLE
watch for changes to config files and automatically restart

### ISSUE THREAD

**Issue by devinrhode2** [sharegate/craco#112]
It'd be very awesome if craco automatically watched for changes to craco.config.js (and maybe other files you set, like .eslintrc.js) and automatically restarted the dev server.

**Comment by patricklafrance** [comment]
Hi @devinrhode2 

I am not sure this is something I want to add to the core of craco.

That could be an addon or instructions could be added to explain how to do that.

What do you think?

Thank you,

Patrick

**Comment by devinrhode2** [comment]
I'm not really familiar writing with webpack plugins at all

Do you think it'd be possible to write a plugin that watches craco.config.js and re-loads changes from that file?

**Comment by devinrhode2** [comment]
I bet there is a simple command that can come before the `craco start` command that could help with this...
Sure enough... using `npm install onchange --save-dev` and set start script to:
```
    "start": "onchange --initial --kill \"craco.config.js\" -- craco start",
```
only drawback is that you don't get console clearing. I however do get colors. But I'm not sure I'm too concerned with console clearing right now.

As far as restarting craco dev server when the craco config changes, I think this is probably case closed. If you agree we can close this.

**Comment by patricklafrance** [comment]
Hi @devinrhode2 

Yes, I had something like this in mind! I think that's a very good way to achieve this.

Thank you, I will close this issue since there isn't any further step to do. That will be usefull information if someone else want to achieve this.

Patrick

**Comment by devinrhode2** [comment]
Just some notes for anyone looking to do something similar:
First, I am on mac - that's important cause watching files ends up working slightly differently on all systems.

I did try to use entr, seemed like a better more standard module for just doing some simple watching. Before that I had done some very broad google searching, and kept coming around to polling solutions (suggesting using the brew package `watch` or `visionmedia-watch` both of which just use polling.)

I struggled to understand how to use watchman cli to watch one file and trigger the command. 

What makes me happy about the onchange utility I found, is that I can install it just like any other npm dependency, and anyone else working on the project doesn't even have to know it's there. They don't even need to know that craco.config.js needs to be restarted when the file is changed - because it restarts itself. After I run with this option for a few weeks or so I will try and post an update here commenting whether or not I have kept this craco config watcher

It was also very easy to figure out the --initial and --kill options needed to make this work. I wish something "more robust" like entr worked but at the end of the day onchange helped me get the job done the fastest with no issues. It could also possibly be more lightweight than watchman, but I'm not sure.
